packages/react-scripts/config/paths.js

```diff
@@ -35,6 +36,24 @@ var nodePaths = (process.env.NODE_PATH || '')
   .filter(Boolean)
   .map(resolveApp);
 
+var envPublicUrl = process.env.PUBLIC_URL;
+
+function getPublicUrl(appPackageJson) {
+  return envPublicUrl ? envPublicUrl : require(appPackageJson).homepage;
+}
+
+// We use `PUBLIC_URL` environment variable or "homepage" field to infer
+// "public path" at which the app is served.
+// Webpack needs to know it to put the right <script> hrefs into HTML even in
+// single-page apps that may serve index.html for nested URLs like /todos/42.
+// We can't use a relative path in HTML because we don't want to load something
+// like /todos/42/static/js/bundle.7289d.js. We have to know the root.
+function getServedPath(appPackageJson) {
```

Comment:
sorry for the tersity.

the middle line could be irrelevant depending on `envPublicUrl`, which comes as a surprise while reading. I think you should branch earlier on `if (envPublicUrl)` to make that clear.

There's also confusion between "public url", "homepage path", and "served path". It seems like this method is saying that the served path is derived from the public url.  So I don't think "homepage" should be in the variable names.